extensions/parquet/base/src/main/java/io/deephaven/parquet/base/util/LocalFSChannelProvider.java

```diff
@@ -14,8 +15,22 @@
 public class LocalFSChannelProvider implements SeekableChannelsProvider {
 
     @Override
-    public SeekableByteChannel getReadChannel(@NotNull final Path path) throws IOException {
-        return FileChannel.open(path, StandardOpenOption.READ);
+    public SeekableChannelContext makeContext() {
+        // No additional context required for local FS
+        return SeekableChannelContext.NULL;
+    }
+
+    @Override
+    public boolean isCompatibleWith(@NotNull final SeekableChannelContext channelContext) {
+        return channelContext == SeekableChannelContext.NULL;
```

Comment:
You can be more permissive than this. The `LocalFSChannelProvider` doesn't use a context, hence all contexts are compatible. There might be utility in this, if you were scanning a `PartitionedAwareSourceTable` with heterogenous regions.
```suggestion
        return true;
```